wordpress.org/public_html/wp-content/plugins/theme-directory/class-wporg-themes-auto-review.php

```diff
@@ -0,0 +1,173 @@
+		$this->namespace         = 'themes/v1';
+		$this->rest_base         = 'github';
+
+		$this->register_routes();
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<theme_slug>[^/]+)/(?P<ticket_id>[\d]+)/',
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * A Permission Check callback which validates the request against a WP_ORG token.
+	 *
+	 * @param \WP_REST_Request $request The Rest API Request.
+	 * @return bool|\WP_Error True if the token exists, WP_Error upon failure.
+	 */
+	function update_item_permissions_check( $request ) {
+		return $this->permission_check_api_bearer( $request, 'AUTO_REVIEW_TRAC_API_GITHUB_BEARER_TOKEN' );
+	}
+
+	/**
+	 * A Permission Check callback which validates the a request against a given token.
+	 *
+	 * @param \WP_REST_Request $request  The Rest API Request.
+	 * @param string           $constant The constant that contains the expected bearer.
+	 * @return bool|\WP_Error True if the token exists, WP_Error upon failure.
+	 */
+	function permission_check_api_bearer( $request, $constant = false ) {
+		$authorization_header = $request->get_header( 'authorization' );
+		$authorization_header = trim( str_ireplace( 'bearer', '', $authorization_header ) );
+
+		if (
+			! $authorization_header ||
+			! $constant ||
+			! defined( $constant ) ||
+			! hash_equals( constant( $constant ), $authorization_header )
+		) {
+			return new \WP_Error(
+				'not_authorized',
+				__( 'Sorry! You cannot do that.', 'wporg-plugins' ),
+				array( 'status' => \WP_Http::UNAUTHORIZED )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns whether the ticket number is for the correct theme.
+	 * 
+	 * @param string $keywords A string that should include the theme slug.
+	 * @param string $theme_slug The theme slug.
+	 * @return bool
+	 */
+	public function ticket_is_for_theme( $keywords, $theme_slug ) {
+		return ! empty( $keywords ) && strpos( $keywords , $theme_slug );
+	}
+
+	/**
+	 * Returns a cleaned up version of the content.
+	 * 
+	 * @param string $content GitHub specific language to format.
+	 * @return string
+	 */
+	public function get_formatted_content( $content ) {
+		$padding = PHP_EOL . PHP_EOL;
+
+		// Add some space around the titles
+		$content = str_replace( "]:", "]:" . $padding, $content );
+		$content = str_replace( "[", $padding . "[", $content );
+
+		// Remove empty spaces
+		$content = trim( $content );
+
+		$disclaimer = sprintf( __( "''The following has been generated by https://github.com/WordPress/theme-review-action, this is for informational purposes only.''" . PHP_EOL, 'wporg-themes' ) );
```

Comment:
The Trac formatting `''` should be moved out of the translated string here.

I'm not sure translating it is of much benefit either though, as it'll probably always be in English. It could be marked for translation later if need be?

Reply:
Yeah fair enough, I'll remove the translation for now since this is experimental.